src/main/java/uk/gov/hmcts/reform/draftstore/service/DraftService.java

```diff
@@ -0,0 +1,88 @@
+package uk.gov.hmcts.reform.draftstore.service;
+
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.draftstore.data.DraftStoreDAO;
+import uk.gov.hmcts.reform.draftstore.domain.CreateDraft;
+import uk.gov.hmcts.reform.draftstore.domain.Draft;
+import uk.gov.hmcts.reform.draftstore.domain.DraftList;
+import uk.gov.hmcts.reform.draftstore.domain.UpdateDraft;
+import uk.gov.hmcts.reform.draftstore.exception.AuthorizationException;
+import uk.gov.hmcts.reform.draftstore.exception.NoDraftFoundException;
+
+import java.util.List;
+import java.util.Objects;
+
+@Service
+public class DraftService {
+
+    private final DraftStoreDAO draftRepo;
+
+    public DraftService(DraftStoreDAO draftRepo) {
+        this.draftRepo = draftRepo;
+    }
+
+    public Draft read(String id, UserAndService userAndService) {
+        return draftRepo
+            .read(toInternalId(id))
+            .filter(draft -> Objects.equals(draft.userId, userAndService.userId))
+            .filter(draft -> Objects.equals(draft.service, userAndService.service))
+            .orElseThrow(() -> new NoDraftFoundException());
+    }
+
+    public DraftList read(UserAndService userAndService, Integer after, int limit) {
```

Comment:
Is there a reason why `after` is of type Integer?

Reply:
paging is optional. the type hasn't changed in this refactoring.